### PR TITLE
[FIX] mrp{_account}: consider capacities when computing expected duration

### PR DESCRIPTION
**Pull request by stcc-odoo**
Related enterprise PR: https://github.com/odoo/enterprise/pull/39669

Steps to reproduce:

- Install mrp_workorder
- Create Product P1, Workcenter W1
- Edit W1 > Specific capacities > Add line: Product = P1, Start time = 5, End time = 5
- Create BoM > Product: P1, Operations tab > Add line: Workcenter = WC1, Default duration = 60:00 min > Save
- Click on overview stat button

Issue:

The created operation has expected time of 60:00, instead of 70:00. The workcenter start and stop times are considered in this calculation, so the workcenter capacities should be considered too.

Solution:

Use `_get_expected_duration` to compute the expected duration.

opw-3229485

### CODE REVIEW

**Review by nle-odoo — APPROVED** [review]
looks good to me if it's what we want

**Comment by stcc-odoo** [comment]
@odoo/rd-logistics :slightly_smiling_face:

**Comment by stcc-odoo** [comment]
@odoo/rd-logistics could you please have a look at this PR? :slightly_smiling_face:

**Comment by ticodoo** [comment]
Fix is good for me, just needs to also be added to the `_compute_bom_price` in product.py https://github.com/odoo/odoo/blob/16.0/addons/mrp_account/models/product.py#L79-L82

File: addons/mrp_account/tests/test_bom_price.py [hunk 1]
```diff
@@ -111,15 +111,24 @@ def test_00_compute_price(self):
...
+        workcenter_1 = workcenter_form1.save()
+
+        capacity_dining_table = self.env['mrp.workcenter.capacity'].create({
+            'product_id': self.dining_table.id,
+            'workcenter_id': workcenter_1.id,
+            'time_start': 2,
+            'time_stop': 1,
+        })
+
+        workcenter_1.write({'capacity_ids': [(4, capacity_dining_table.id)]})
```
Comment (ticodoo) [inline addons/mrp_account/tests/test_bom_price.py:131]: This line:

`workcenter_1.write({'capacity_ids': [(4, capacity_dining_table.id)]})`

is redundant with the `'workcenter_id': workcenter_1.id,` in the `create`

File: addons/mrp/tests/test_bom.py [hunk 2]
```diff
@@ -599,6 +599,15 @@ def test_20_bom_report(self):
...
         })
 
+        capacity_cheese_cake = self.env['mrp.workcenter.capacity'].create({
+            'product_id': cheese_cake.id,
+            'workcenter_id': workcenter_2.id,
+            'time_start': 2,
+            'time_stop': 1,
+        })
+
+        workcenter_2.write({'capacity_ids': [(4, capacity_cheese_cake.id)]})
```
Comment (ticodoo) [inline addons/mrp/tests/test_bom.py:609]: This line:

`workcenter_2.write({'capacity_ids': [(4, capacity_cheese_cake.id)]})`

is redundant with the `'workcenter_id': workcenter_2.id,` in the `create`

**Review by ticodoo — COMMENTED** [review]
Thanks for the changes, it's super close now! Just a couple nitpick cleanups for your test + for your commit header, since it's only 2 modules that overlap in name you can write it as:

[FIX] mrp{_account}: consider capacities when computing duration

**Comment by ticodoo** [comment]
thanks!

@robodoo r+

**Pull request closed without merging**